### ISSUE TITLE
How to know a if a multiset is subset of another multiset 

### ISSUE THREAD

**Issue by lgutie16** [Yomguithereal/mnemonist#88]
I'm trying to find a way to if a multiset is subset of another multiset 

example:  
letters = {  a, a, a, b, c, d, d  }
lessLetters = {  a, a, b, c }
secondLessLetters = {  z, k }

lessLetters.isSubsetOf(letters) // true 
secondLessLetters.isSubsetOf(letters) // false

Is there a way to get that result?

this is my approach

```
function isSubset(a, b){
    subset =  true
    a.forEach(function(value) {
        subset = (b.multiplicity(value) > a.multiplicity(value)) && subset
    });
    return subset
}


isSubset(lessLetters, letters) // true
isSubset(secondLessLetters, letters) // false
```
Thanks,

**Comment by Yomguithereal** [comment]
Hello @lgutie16. Your approach should work indeed. You can optimize slightly though.

I therefore added the functions as static of the MultiSet module. The `isSubset` function looks like that:

```js
function(A, B) {
  var iterator = A.multiplicities(),
      step,
      key,
      mA;

  // Shortcuts
  if (A === B)
    return true;

  if (A.dimension > B.dimension)
    return false;

  while ((step = iterator.next(), !step.done)) {
    key = step.value[0];
    mA = step.value[1];

    if (B.multiplicity(key) < mA)
      return false;
  }

  return true;
}
```

Do you want me to release a version on npm with the newly added methods?

**Comment by Yomguithereal** [comment]
Just published v0.23.0 with the added methods if you need them.
